update the MHA to enable cudnn-frontend 1.12.1

cicichen01 opened the pull request:
Summary:
As titled.
Before: 
```
fbcode/caffe2/aten/src/ATen/native/cudnn/MHA.cpp:490:12: error: no member named 'set_generate_stats' in 'cudnn_frontend::graph::SDPA_attributes'
  488 |       fe::graph::SDPA_attributes()
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  489 |           .set_name("CUDNN_SDPA")
      |           ~~~~~~~~~~~~~~~~~~~~~~~
  490 |           .set_generate_stats(return_softmaxstats)
      |            ^
fbcode/caffe2/aten/src/ATen/native/cudnn/MHA.cpp:708:12: error: no member named 'set_generate_stats' in 'cudnn_frontend::graph::SDPA_attributes'
  706 |       fe::graph::SDPA_attributes()
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  707 |           .set_name("CUDNN_SDPA_NESTEDTENSOR")
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  708 |           .set_generate_stats(return_softmaxstats)
      |            ^
2 errors generated.
```

Test Plan: sandcastle

Differential Revision: D87832011

henrylhtsang commented:
This looks good to me. If no objection I will stamp 

cc @drisspg, @zhuhan0, @Skylion007 

from https://github.com/pytorch/pytorch/pull/165776#issuecomment-3444261279

henrylhtsang commented:
> This looks good to me. If no objection I will stamp
> 
> cc @drisspg, @zhuhan0, @Skylion007
> 
> from [#165776 (comment)](https://github.com/pytorch/pytorch/pull/165776#issuecomment-3444261279)

cc @eqy

eqy commented on aten/src/ATen/native/cudnn/MHA.cpp:
```diff
@@ -487,7 +487,11 @@ std::unique_ptr<fe::graph::Graph> build_graph(
       fe::graph::SDPA_attributes()
           .set_name("CUDNN_SDPA")
+#if defined(CUDNN_FRONTEND_VERSION) && CUDNN_FRONTEND_VERSION == 11200
+          .set_is_inference(!return_softmaxstats)
```
This should be `CUDNN_FRONTEND_VERSION <= 11200`, and no need to use `defined(CUDNN_FRONTEND_VERSION)` as this entire block should be enclosed by that

cicichen01 commented:
@keep-going

cicichen01 commented:
@pytorchbot rebase

cicichen01 committed:
update the MHA to enable cudnn-frontend 1.12.1 (#169086)

Summary:
As titled.
Before: 
```
fbcode/caffe2/aten/src/ATen/native/cudnn/MHA.cpp:490:12: error: no member named 'set_generate_stats' in 'cudnn_frontend::graph::SDPA_attributes'
  488 |       fe::graph::SDPA_attributes()
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  489 |           .set_name("CUDNN_SDPA")
      |           ~~~~~~~~~~~~~~~~~~~~~~~
  490 |           .set_generate_stats(return_softmaxstats)
      |            ^
fbcode/caffe2/aten/src/ATen/native/cudnn/MHA.cpp:708:12: error: no member named 'set_generate_stats' in 'cudnn_frontend::graph::SDPA_attributes'
  706 |       fe::graph::SDPA_attributes()
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  707 |           .set_name("CUDNN_SDPA_NESTEDTENSOR")
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  708 |           .set_generate_stats(return_softmaxstats)
      |            ^
2 errors generated.
```


Test Plan: sandcastle

Reviewed By: henrylhtsang

Differential Revision: D87832011

Pulled By: cicichen01
